### ISSUE TITLE
d3rlpy model-based rl - Tensor related Error

### ISSUE THREAD

**Issue by MishraIN** [takuseno/d3rlpy#181]
Hey, I am getting error using custom env.  I am following the code here, https://d3rlpy.readthedocs.io/en/v1.0.0/references/model_based.html
I am getting error in this part of the code, "Tensors must have same number of dimensions: got 2 and 1". How can fix this issue? 

 `dynamics.fit(train_episodes,
             eval_episodes=test_episodes,
             n_epochs=100,
             scorers={
                'observation_error': dynamics_observation_prediction_error_scorer,
                'reward_error': dynamics_reward_prediction_error_scorer,
                'variance': dynamics_prediction_variance_scorer,
             })`

My action and observation space for custom env is ,
Action Space: Box(0.0, 100.0, (1,), float32)
Observation Space: Box(0.0, 100.0, (1,), float32)
Thanks!

**Comment by takuseno** [comment]
Thanks for the issue. What's your `dynamics` setting here? Also, that would be very helpful if you share the example actions and observations.

**Comment by MishraIN** [comment]
For creating dataset
`def play_game(env, num_games_for_dataset = 2500):
    
    observation_list = []
    action_list = []
    reward_list = []
    terminal_list = []
    timeout_list = [] #THIS WILL ALWAYS BE 0 BECAUSE I DON"T HAVE A MAX EPISODE LENGTH
    info_list = [] ## This is also going to be empty

    for i in np.arange(num_games_for_dataset):
        done = False
        env.reset()
        # print('start')
        while done == False: # run until game over
            #env.render() #Don't Render the environment
            action = env.action_space.sample()
            obs, reward, done, info= env.step(action)
            # print(obs)

            observation_list.append(obs)
            action_list.append(action)
            reward_list.append(reward)
            terminal_list.append(done)
            timeout_list.append(False)
            info_list.append(info)
    return observation_list, action_list, reward_list, terminal_list

observations, actions, rewards, terminals = play_game(env, num_games_for_dataset=10)
dataset = MDPDataset(np.asarray(observations), np.asarray(actions).reshape(-1,1), np.asarray(rewards), np.asarray(terminals))`

dynamics setting 
`dynamics = d3rlpy.dynamics.ProbabilisticEnsembleDynamics(learning_rate=1e-4, use_gpu=False)`

I am using this game 
[https://medium.com/cloudcraftz/build-a-custom-environment-using-openai-gym-for-reinforcement-learning-56d7a5aa827b]

Thank you!

**Comment by takuseno** [comment]
One possibility is that the `actions` include only integer values (it could be float, but no decimals). In that case, `MDPDataset` automatically interprets the `actions` as discrete action-space. You can probably try setting `discrete_action=False` at `MDPDataset`'s argument.

**Comment by MishraIN** [comment]
Thanks! I was a step further this time but using a different game.

action space : Box([-1.], [1.], (1,), float32)

observation space : Box([-4.8000002e+00 -3.4028235e+38 -4.1887903e-01
-3.4028235e+38], [4.8000002e+00 3.4028235e+38 4.1887903e-01
3.4028235e+38], (4,), float32)


I don't understand the output from this code "2022-04-27 15:44.35
[warning  ] Unused arguments are passed.
algorithm=ProbabilisticEnsembleDynamics gamma=1.0
generated_maxlen=100000 n_steps=1 real_ratio=1.0"


<from d3rlpy.algos import MOPO

# load trained dynamics model
dynamics = ProbabilisticEnsembleDynamics.from_json(json)
dynamics.load_model(model)

# give mopo as generator argument.
mopo = MOPO(dynamics=dynamics)>


Thanks!

**Comment by takuseno** [comment]
That's the just warning. You can ignore them. Sounds like the issue is solved now. Please let me close this issue, but you can freely reopen this if necessary.
